app/models/common_tagging.rb

```diff
@@ -30,7 +33,8 @@ def inherit_parents
     end
   end
 
+  # If a tag's parent changes, reindex immediately to update unwrangled bins.
   def update_search
```

Comment:
It might be better to have this as an after_create + after_destroy callback, instead of after_commit. But it's a tradeoff between handling stale data better vs. better error handling, so it's mostly a judgment call.